Source/WPEFramework/PluginServer.cpp

```diff
@@ -709,12 +709,22 @@ namespace PluginHost {
                 result = Core::ERROR_BAD_REQUEST;
             }
             else {
+                State(IShell::HIBERNATED);
 #ifdef HIBERNATE_SUPPORT_ENABLED
                 Core::process_t parentPID = local->ParentPID();
+                local->Release();
+                Unlock();
 
                 SYSLOG(Logging::Notification, ("Hibernation of plugin [%s] process [%u]", Callsign().c_str(), parentPID));
                 result = HibernateProcess(timeout, parentPID, _administrator.Configuration().HibernateLocator().c_str(), _T(""), &_hibernateStorage);
-                if(result == HIBERNATE_ERROR_NONE)
+                Lock();
+                if (State() != IShell::HIBERNATED) {
+                    SYSLOG(Logging::Startup, ("Hibernation aborted of plugin [%s] process [%u]", Callsign().c_str(), parentPID));
```

Comment:
This is good as it tells the operator that is action to hibernate has aborted (see category Logging::startup) so perfect !